### ISSUE TITLE
Intermittent crashes on macOS High Sierra

### ISSUE THREAD

**Issue by roele** [felixhageloh/uebersicht#289]
Übersicht (version 1.2 build 48) crashes intermittently on macOS High Sierra. See [Übersicht_2017-10-02-195121.log](https://github.com/felixhageloh/uebersicht/files/1350008/Ubersicht_2017-10-02-195121.log) for more details about the crash.

**Comment by felixhageloh** [comment]
I think it always/only happens when you plug in or unplug a screen. Do you observe the same?

**Comment by roele** [comment]
I am using a MacBook Pro without external screen. So far i could not observe when exactly the crash happens but if i'm able to i let you know.

**Comment by roele** [comment]
Here's an exerpt from the `system.log` after a crash.
```
Oct  7 08:30:00 Rio-de-Janeiro com.apple.xpc.launchd[1] (com.apple.newsyslog): Service only ran for 5 seconds. Pushing respawn out by 5 seconds.
Oct  7 08:30:00 Rio-de-Janeiro SophosWebD[382]: [WebClient.m:384] starting intercept
Oct  7 08:30:00 Rio-de-Janeiro SophosSXLD[367]: [SMESXLInterface.m:436] no DNS addresses found
Oct  7 08:30:00 Rio-de-Janeiro SophosWebD[382]: [WebClient.m:384] starting intercept
Oct  7 08:30:01 Rio-de-Janeiro SophosAntiVirus[371]: connected to Sophos Server
Oct  7 08:30:01 Rio-de-Janeiro xpcproxy[24382]: libcoreservices: _dirhelper_userdir: 529: bootstrap_look_up returned (ipc/send) invalid destination port
Oct  7 08:30:01 Rio-de-Janeiro netbiosd[24382]: DEPRECATED USE in libdispatch client: Setting timer interval to 0 requests a 1ns timer, did you mean FOREVER (a one-shot timer)?
Oct  7 08:30:06 Rio-de-Janeiro iTunesHelper[569]: Entered:_AMMuxedDeviceDisconnected, mux-device:17
Oct  7 08:30:06 Rio-de-Janeiro iTunesHelper[569]: Entered:__thr_AMMuxedDeviceDisconnected, mux-device:17
Oct  7 08:30:06 Rio-de-Janeiro iTunesHelper[569]: tid:7fe3 - Mux ID not found in mapping dictionary
Oct  7 08:30:06 Rio-de-Janeiro iTunesHelper[569]: tid:7fe3 - Can't handle disconnect with invalid ecid
Oct  7 08:30:23 Rio-de-Janeiro timed[98]: settimeofday({0x59d8747f,0xebbed}) == 0
Oct  7 08:30:28 Rio-de-Janeiro iTunesHelper[569]: Entered:_AMMuxedDeviceDisconnected, mux-device:18
Oct  7 08:30:28 Rio-de-Janeiro iTunesHelper[569]: Entered:__thr_AMMuxedDeviceDisconnected, mux-device:18
Oct  7 08:30:28 Rio-de-Janeiro iTunesHelper[569]: tid:7fe3 - Mux ID not found in mapping dictionary
Oct  7 08:30:28 Rio-de-Janeiro iTunesHelper[569]: tid:7fe3 - Can't handle disconnect with invalid ecid
Oct  7 08:30:28 Rio-de-Janeiro com.apple.xpc.launchd[1] (com.apple.imfoundation.IMRemoteURLConnectionAgent): Unknown key for integer: _DirtyJetsamMemoryLimit
Oct  7 08:30:28 Rio-de-Janeiro com.apple.xpc.launchd[1] (tracesOf.Uebersicht.3928[16054]): Service exited due to signal: Segmentation fault: 11 sent by exc handler[0]
Oct  7 08:30:28 Rio-de-Janeiro com.apple.xpc.launchd[1] (com.apple.WebKit.Networking.58F26BDB-5A1F-4226-9BDC-146094953B07[16058]): Service exited with abnormal code: 1
```
You might be right, it looks this is related to USB connectivity.

**Comment by roele** [comment]
Another crash but this time without prior device connection errors. Seems related to **com.apple.imfoundation.IMRemoteURLConnectionAgent** which also occurs in the previous log.

```
**Oct  8 13:37:32 Rio-de-Janeiro com.apple.xpc.launchd[1] (com.apple.imfoundation.IMRemoteURLConnectionAgent): Unknown key for integer: _DirtyJetsamMemoryLimit**
Oct  8 13:37:32 Rio-de-Janeiro com.apple.xpc.launchd[1] (tracesOf.Uebersicht.5692[18066]): Service exited due to signal: Segmentation fault: 11 sent by exc handler[0]
```

**Comment by felixhageloh** [comment]
Thanks again! I have also seen it crash in other situations than just when changing screens now.
From the crash log itself it looks like it is a webview bug btw, but maybe a bug I can avoid triggering by changing some behavior of Uebersicht

**Comment by Jarvvski** [comment]
Currently it won't start up at all for me:

<img width="696" alt="image" src="https://user-images.githubusercontent.com/6012840/31992600-9ff96028-b972-11e7-910d-94cc271461bb.png">

Closing and re-launching just gets it to the same `not responding` state as it runs up the hangs (in the time it took me to write this short message it's gone up to 56 hangs from 40)

**Comment by felixhageloh** [comment]
I think I found and fixed the crash in v 1.2.53 (released just now).

@Jarvvski I am not sure if your issue is related. Does it still happen with 1.2.53?
